### PR TITLE
Npm publish

### PR DESCRIPTION
**Pull request by Adamik10**
#### What does this PR do?
- Introduces the ability to create an NPM package out of the repository that can be used for styling and assets in other projects 
- In connection to this, the README.md is also adjusted for documentation of this feature
- Some Github actions have also been altered in connection to this 

#### Where should the reviewer start?
- Just file by file 

#### How can the reviewer see this in action?
-

#### Any background context you want to provide?
The package functionality was conceived to tackle the problem with another repository - ddb-react  where we previously had to copy paste whole css files in order to take advantage of the styling from this repository. 

#### Is this related to processing of personal data?
No.

#### What are the relevant tickets?
N/A

#### Screenshots (if appropriate)
-

#### Questions for the reviewer:
-

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,27 +1,8 @@
...
-    "react-docgen-typescript": "^2.1.0",
-    "react-dom": "^17.0.2",
-    "react-scripts": "4.0.3",
-    "typescript": "^4.1.2",
-    "web-vitals": "^1.0.1"
-  },
-  "engines": {
-    "node": ">=16",
-    "npm": ">=7"
-  },
```
Comment (kasperg) [inline package.json:24]: The commit comment for this change is:

>  Test
> - removal of engines setting in package.json and adding .nvmrc
> - we are doing this in order to be compatible with the node version of ddb-react

Next time consider improving the headline and adding a few more details about your thought process here. A suggestion might be:

>  Remove of engines setting in package.json
>
> Package managers will not allow installation of packages which require a newer engine than what is available. ddb-react currently requires Node 12 which conflicts with the current package and will prevent installation. Even though this package only requires Node 16 for development NPM & Co has no way to distinguish between development and runtime requirements.
>
> To resolve the situation we end up removing the engine specification from package.json. To compensate we add Node version requirements to `.nvmrc` so it will be picked up more easily here. Otherwise we just have to hope developers follow the README.

Perhaps a bit wordy but the important parts are:

1. Getting in the headline right
2. Making clear the reasons why you made the change in the first place—the way things worked before the change (and what was wrong with that), the way they work now, and why you decided to solve it the way you did

https://cbea.ms/git-commit/#why-not-how is a great resource about commit messages :smile:

**Review by kasperg — APPROVED** [review]
:+1: but that almost goes without saying. The code is basicly what we have already agreed on in #1 😄 .

File: package.json (hunk 1 above)
Comment (Adamik10) [reply package.json:24]: Thank you @kasperg, I amended this and I shall do better with commit messages in the future.

**Comment by Adamik10** [comment]
@spaceo @kasperg what are these "UI Tests"?
image

**Comment by kasperg** [comment]
@Adamik10 These are Chromatic tests which will highlight UI changes based on the code changes.

Chromatic has a review flow with allows us to accept or reject changes which we may want to use or not.

**Review by spaceo — APPROVED** [review]

File: .github/workflows/deployment.yml [hunk 2]
```diff
@@ -29,12 +25,63 @@ jobs:
...
+        if: startsWith(github.ref, 'refs/heads/')
+        # Version 0.0.0-SHA is a schema that supports semantic versioning but
+        # should sink below proper versions.
+        # Output package.json to provide insight and help debugging
+        # Using branch names as tags allows other projects to track unreleased
+        # development.
+        run: |
+          npm version 0.0.0-$GITHUB_SHA --no-git-tag-version
+          cat package.json
+          npm publish --tag $GITHUB_REF_NAME
```
Comment (kasperg) [inline .github/workflows/deployment.yml]: Branch names may contain special characters such as `/` which will not work as tag names for NPM packages. We talked about using a GitHub Action for this but perhaps it can be solved using a little Bash instead?


Pseudo code:

```
$NPM_TAG_NAME = replace("/", "-", $GITHUB_REF_NAME)
npm publish --tag $NPM_TAG_NAME
```

Comment (kasperg) [reply .github/workflows/deployment.yml]: Working code:

```bash
 NPM_TAG_NAME=$(echo $GITHUB_REF_NAME | tr -d [:space:] | tr -C [:alnum:] _)
 npm publish --tag $NPM_TAG_NAME
 ```
